### PR TITLE
'istioctl proxy-status': keep going even if a DiscoveryPod fails proxyGet()

### PR DESCRIPTION
**Pull request by esnible**
Resolves https://github.com/istio/istio/issues/26300

### CODE REVIEW

File: pkg/kube/client.go [hunk 1]
```diff
@@ -479,17 +479,23 @@ func (c *client) AllDiscoveryDo(ctx context.Context, istiodNamespace, path strin
...
-			return nil, err
+			err = multierror.Append(errs, err)
+			continue
 		}
 		if len(res) > 0 {
 			result[istiod.Name] = res
 		}
 	}
-	return result, err
+	// If any Discovery servers responded, treat as a success
```
Comment (howardjohn) [inline pkg/kube/client.go:494]: should we give an warning or something still? might be confusing otherwise

Comment (esnible) [reply pkg/kube/client.go:495]: fixed

**Review by tanjunchen — APPROVED** [review]
/lgtm

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**